flight/Modules/Battery/battery.c

```diff
@@ -128,18 +128,28 @@ static void batteryTask(void * parameters)
 				currentADCPin = -1;
 		}
 
+		bool adc_pin_invalid = false;
+		bool adc_offset_invalid = false;
+
 		// handle voltage
 		if (voltageADCPin >= 0) {
 			flightBatteryData.Voltage = (float)PIOS_ADC_GetChannelVolt(voltageADCPin);
 
 			// A negative result indicates an error (PIOS_ADC_GetChannelVolt returns -1 on error)
 			if(flightBatteryData.Voltage < 0.0f)
-				flightBatteryData.Voltage = 0.0f;
+				adc_pin_invalid = true;
```

Comment:
I think it might be cleaner to do

```
// A negative result indicates an error (PIOS_ADC_GetChannelVolt returns -1 on error)
if(flightBatteryData.Voltage < 0.0f) {
	flightBatteryData.Voltage = 0.0f;
	adc_pin_invalid = true;
} else {
	// scale to actual voltage
	flightBatteryData.Voltage *= 1000.0f / batterySettings.SensorCalibrationFactor[FLIGHTBATTERYSETTINGS_SENSORCALIBRATIONFACTOR_VOLTAGE];
	flightBatteryData.Voltage += batterySettings.SensorCalibrationOffset[FLIGHTBATTERYSETTINGS_SENSORCALIBRATIONOFFSET_VOLTAGE]; //in Volts
}
```